tiledb/sm/array_schema/domain.cc

```diff
@@ -245,10 +229,11 @@ int Domain::cell_order_cmp(
     return 1;
   }
 
-  assert(cell_order_cmp_func_2_[dim_idx] != nullptr);
-  auto coord_a = a.coord(dim_idx);
-  auto coord_b = b.coord(dim_idx);
-  return cell_order_cmp_func_2_[dim_idx](coord_a, coord_b);
+  if (cell_order_cmp_func_2_[dim_idx] == nullptr) {
+    assert(cell_order_cmp_func_2_[dim_idx] != nullptr);
+    return 0;  // WORKAROUND: Mask the defect by returning instead of throwing
```

Comment:
It's a change in behavior, but only in release builds. If the function is missing, `assert` will terminate the program. I wrote this code with an `if` statement rather than a raw `assert` so that it could readily be changed to `throw std::logic_error` (lacking our own exception object as yet).

And really, it's only a change in the way this code copes with defects elsewhere in the code base. I had considered fixing the code so that there were no `nullptr` initializations of these functions at all, but that would have been scope creep even farther than this PR already had. And it's a systemic problem, because we have `Datatype` values defined that (1) we don't fully support, but (2) allow data to be created with. Just getting rid of null functions doesn't actually solve the problem, even if it gets rid of the problem right here.

I wrote this code before we had any of the exception safety wrappers merged in. I would prefer `throw` here when this merges, slightly prior to finishing exception safety in the API, but I'm fairly agnostic about the interim.

Reply:
I would lean towards maintaining the current behavior and crash, rather than potentially giving incorrect results. @stavrospapadopoulos @Shelnutt2 what do you think?